pfhedge/stochastic/rough_bergomi.py

```diff
@@ -0,0 +1,147 @@
+          Code: https://github.com/ryanmccrickerd/rough_bergomi
+
+    Args:
+        n_paths (int): The number of simulated paths.
+        n_steps (int): The number of time steps.
+        init_state (tuple[torch.Tensor | float], optional): The initial state of
+            the time series.
+            This is specified by a tuple :math:`(S(0), V(0))`.
+            If ``None`` (default), it uses :math:`(1.0, \\xi)`.
+        alpha (float, default=-0.4): The parameter :math:`\\alpha`.
+        rho (float, default=-0.9): The parameter :math:`\\rho`.
+        eta (float, default=1.9): The parameter :math:`\\eta`.
+        xi (float, default=0.04): The parameter :math:`\\xi`.
+        dt (float, default=1 / 250): The intervals of the time steps.
+        dtype (torch.dtype, optional): The desired data type of returned tensor.
+            Default: If ``None``, uses a global default
+            (see :func:`torch.set_default_tensor_type()`).
+        device (torch.device, optional): The desired device of returned tensor.
+            Default: If ``None``, uses the current device for the default tensor type
+            (see :func:`torch.set_default_tensor_type()`).
+            ``device`` will be the CPU for CPU tensor types and the current CUDA device
+            for CUDA tensor types.
+
+    Shape:
+        - spot: :math:`(N, T)` where
+          :math:`N` is the number of paths and
+          :math:`T` is the number of time steps.
+        - variance: :math:`(N, T)`.
+
+    Returns:
+        (torch.Tensor, torch.Tensor): A namedtuple ``(spot, variance)``.
+
+    Examples:
+        >>> from pfhedge.stochastic import generate_rough_bergomi
+        ...
+        >>> _ = torch.manual_seed(42)
+        >>> outputs = generate_rough_bergomi(2, 5)
+        >>> outputs.spot
+        tensor([[1.0000, 0.9807, 0.9563, 0.9540, 0.9570],
+                [1.0000, 1.0147, 1.0097, 1.0107, 1.0164]])
+        >>> outputs.variance
+        tensor([[0.0400, 0.3130, 0.0105, 0.0164, 0.0068],
+                [0.0400, 0.0396, 0.0049, 0.0064, 0.0149]])
+
+    """
+
+    if init_state is None:
+        init_state = (1.0, xi)
+
+    init_state = cast_state(init_state, dtype=dtype, device=device)
+    alpha_tensor, rho_tensor, eta_tensor = cast_state(
+        (alpha, rho, eta), dtype=dtype, device=device
+    )
+
+    _dW1_cov1 = dt ** (alpha + 1) / (alpha + 1)
+    _dW1_cov2 = dt ** (2 * alpha + 1) / (2 * alpha + 1)
+    _dW1_covariance_matrix = torch.as_tensor(
+        [[dt, _dW1_cov1], [_dW1_cov1, _dW1_cov2]], dtype=dtype, device=device
+    )
+    _dW1_generator = torch.distributions.multivariate_normal.MultivariateNormal(
+        loc=torch.as_tensor([0.0, 0.0], dtype=dtype, device=device),
+        covariance_matrix=_dW1_covariance_matrix,
+    )
+
+    dW1 = _dW1_generator.sample(torch.Size([n_paths, n_steps - 1]))
+    dW2 = torch.randn([n_paths, n_steps - 1], dtype=dtype, device=device)
+    dW2 *= dW2.new_tensor(dt).sqrt()
```

Comment:
このケースは使いまわしはなさそうではあるので安全ではあるがレポジトリ全体として in-place は極力避けた方が良いと思います。

```suggestion
    dW2 = dW2 * math.sqrt(dt)
```